### PR TITLE
hmcl: update to 3.5.7

### PR DESCRIPTION
**Pull request by SkyBird233**
Topic Description
-----------------

- hmcl: update to 3.5.7
    Other changes:
    - Use release version instead of develop version
    - Add `Noto Sans CJK SC` as default font
    - Run at $HOME instead of current directory

Package(s) Affected
-------------------

- hmcl: 3.5.7

Security Update?
----------------

No

Build Order
-----------

```
#buildit hmcl
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] Architecture-independent `noarch`

### CODE REVIEW

File: app-games/hmcl/autobuild/build [hunk 1]
```diff
@@ -1,9 +1,15 @@
+abinfo "Building hmcl.jar..."
+export VERSION_TYPE=stable
+export BUILD_NUMBER=$(echo $PKGVER | cut -d '.' -f 3)
+export MICROSOFT_AUTH_ID=81a207c0-a53c-46a3-be07-57d2b28c1643
+export CURSEFORGE_API_KEY="$2a$10$gWtfk4i6mQvxSLhtMdvBfuM1E46vEbmYV4w4sHSJBX6c3hO6WCDWG"
+./gradlew build
```
Comment (MingcongBai) [inline app-games/hmcl/autobuild/build]: "$SRCDIR"/gradlew

File: app-games/hmcl/autobuild/build [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
+ls "${SRCDIR}"/HMCL/build/libs/ | grep ".jar$"
+cp -Lv "${SRCDIR}"/HMCL/build/libs/HMCL*.jar "${PKGDIR}/usr/share/java/hmcl/hmcl.jar"
 
-abinfo "Extracting desktop icon from jar..."
+abinfo "Extracting desktop icon..."
 mkdir -pv "${PKGDIR}"/usr/share/pixmaps
-unzip -p "${PKGDIR}/usr/share/java/hmcl/hmcl.jar" \
-      assets/img/icon\@8x.png \
-      > "${PKGDIR}"/usr/share/pixmaps/hmcl.png
+cp "${SRCDIR}"/HMCL/image/hmcl.png "${PKGDIR}"/usr/share/pixmaps/hmcl.png
```
Comment (MingcongBai) [inline app-games/hmcl/autobuild/build]: Merge the above two commands as follows:

```bash
install -Dvm644 "$SRCDIR"/HCML/image/hmcl.png \
    "$PKGDIR"/usr/share/pixmaps/hmcl.png
```

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-games/hmcl/autobuild/build (hunk 2 above)
Comment (SkyBird233) [reply app-games/hmcl/autobuild/build]: `"$SRCDIR"/HCML/...` should be `"$SRCDIR"/HMCL/...`  x

**Comment by SkyBird233** [comment]
- [x] CurseForge:
CurseForge
- [x] Microsoft login:
Microsoft login

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**